master/internal/db/static_query_map.go

```diff
@@ -0,0 +1,26 @@
+package db
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/determined-ai/determined/master/pkg/etc"
+)
+
+type staticQueryMap struct {
+	queries map[string]string
+	sync.RWMutex
+}
+
+func (q *staticQueryMap) getOrLoad(queryName string) string {
+	q.RLock()
+	query, ok := q.queries[queryName]
+	q.RUnlock()
```

Comment:
question: should we be using `defer` to avoid situations where a panic of some sort happens, causing the unlock to never happen?  (the question applies here and below)

Reply:
If we `defer q.RUnlock()` then I think if `!ok` we could have a deadlock trying to `q.Lock()`. We could `defer q.Unlock()`, but I don't see any way a panic could happen (since we're using the `ok` return from the map).